### ISSUE TITLE
When exiftool process is killed, readMetadata hangs

### ISSUE THREAD

**Issue by mika-fischer** [Sobesednik/node-exiftool#32]
I think the problem might be a race between the the process exiting and the exit listener being installed. Ideally, the exit listener should be installed in the same tick as the `child_process.spawn`, then this could never happen.

I'm basically doing (for a lot of files):
```
const etp = new exiftool.ExiftoolProcess(exiftoolBin);
await etp.open();
const metadata = await etp.readMetadata(filename); // This hangs if exiftool process is killed
```

and on the terminal:
```
taskkill /F /IM exiftool.exe
```

I do realize now that I don't actually need to spawn a new process for each image, so I'll see if I can work around it that way. This is still worth fixing though...

**Comment by z-vr** [comment]
OK thanks for that. [There's a test](https://github.com/Sobesednik/node-exiftool/blob/master/test/spec/exiftool.js#L106) to make sure all remaining jobs are finished when closing properly (with close method), but you're right I haven't considered what happens when a process is killed.

**Comment by z-vr** [comment]
@mika-fischer Fixed in `2.3.0`, thanks for reporting.
